index.js

```diff
@@ -2,52 +2,36 @@
 var through = require('through-gulp')
 var PluginError = require('plugin-error')
 var replaceExt = require('replace-ext')
-var asciidoctor = require('asciidoctor')()
-
-/**
- * Test is the given object is a class
- * @param {object} obj The object to test
- */
-function isClass (obj) {
-  return typeof obj === 'function' && obj.name !== 'Object'
-}
+var asciidoctor = require('@asciidoctor/core')()
 
 module.exports = function (theOptions = {}) {
   var options = theOptions || {}
-  var asciidoctorOptions = {}
+  var asciidoctorOptions = { ...theOptions }
 
   // default config
   var extension = options.extension || '.html'
-  // var extensionRegistry = asciidoctor.Extensions
 
   // AsciiDoctor options
   asciidoctorOptions.safe = options.safe || 'unsafe' // unsafe, safe, server or secure
   asciidoctorOptions.doctype = options.doctype || 'article' // book,inline
   asciidoctorOptions.attributes = options.attributes || ['showtitle']
   asciidoctorOptions.backend = options.backed || 'html5' // defaults to html5
   asciidoctorOptions.doctype = options.doctype || 'article' // defaults to article
-  asciidoctorOptions.header_footer = (options.header_footer === undefined
-    ? true : options.header_footer)
-
-  // If user overrides extension registry, use his registry
-  // if (options.extension_registry !== undefined) {
-  //   extensionRegistry = options.extension_registry
-  // }
 
-  // Load converter if option is set
-  if (options.converter !== undefined) {
-    var cnv = options.converter
-    if (isClass(options.converter)) {
-      const CnvClass = options.converter
-      cnv = new CnvClass()
-    }
-
-    if (typeof cnv.convert === 'function') {
-      asciidoctor.ConverterFactory.register(cnv, asciidoctorOptions.backend)
```

Comment:
I think it can still be useful to pass a custom converter. Any particular reason to remove this feature?

Reply:
No, I don't have any particular reason. I just thought that the templates mechanism will make that obsolete.
I can include it again if you think it is useful